src/controllers/room.ts

```diff
@@ -0,0 +1,40 @@
+import { Request, Response } from 'express';
+import Room from '../models/Room';
+
+export async function createRoom(req: Request, res: Response) {
+    const { name, schedule, closed } = req.body;
+    if (typeof name === 'string' && typeof closed === 'boolean' && Array.isArray(schedule)) {
+        for (const day in schedule) {
+            if (typeof day !== 'string') {
+                console.log(typeof day);
+                return res.status(400).json({ err: 'validation failed' });
+            }
+        }
+        const room = new Room({
+            name,
+            schedule,
+            closed,
+        });
+        await room.save().catch((err) => {
+            return res.status(400).json({ err });
+        });
+        res.status(201).json({ room });
+    } else {
```

Comment:
This else could be refactored to be inverted such that it returns early if the input is wrong, reducing some of the nesting and making the flow easier to read